server/routers/api-router.js

```diff
@@ -139,11 +143,30 @@ router.get("/api/badge/:id/status", cache("5 minutes"), async (request, response
             badgeValues.color = badgeConstants.naColor;
         } else {
             const heartbeat = await Monitor.getPreviousHeartbeat(requestedMonitorId);
-            const state = overrideValue !== undefined ? overrideValue : heartbeat.status === 1;
-
-            badgeValues.label = label ? label : "";
-            badgeValues.color = state ? upColor : downColor;
-            badgeValues.message = label ?? state ? upLabel : downLabel;
+            const state = overrideValue !== undefined ? overrideValue : heartbeat.status;
+
+            badgeValues.label = label ?? "";
+            switch (state) {
+                case 0:
+                    badgeValues.color = downColor;
+                    badgeValues.message = downLabel;
+                    break;
+                case 1:
+                    badgeValues.color = upColor;
+                    badgeValues.message = upLabel;
+                    break;
+                case 2:
+                    badgeValues.color = pendingColor;
+                    badgeValues.message = pendingLabel;
+                    break;
+                case 3:
```

Comment:
Can you use our key names? (PENDING, UP etc)

Reply:
Good point, but this is merged already I'm not sure how should I handle this.

Reply:
New PR ;)